### ISSUE TITLE
Views, referencing another database crash with: `1105 - exceeded max analysis iterations (8)`

### ISSUE THREAD

**Issue by theme1256** [dolthub/dolt#7223]
I'm trying to migrate a system to dolt, but I'm having issues with the views.  
The views are created, but whenever I try to query them I get `1105 - exceeded max analysis iterations (8)`.

Here's a basic example:  
Database 1: "base"
Table: "version"
| VersionMetadata | VersionDatabase | Date |
|--------|--------|--------|
| 1 | 171 | 2012-04-16 | 

Database 2: "live"  
Trying to create view: "version"
| VersionMetadata | VersionDatabase | Date |
|--------|--------|--------|
| 1 | 171 | 2012-04-16 | 

```
create view version as
select
`base`.`version`.`VersionMetadata` AS `VersionMetadata`,
`base`.`version`.`VersionDatabase` AS `VersionDatabase`,
`base`.`version`.`Date` AS `Date`
from `base`.`version`
```
Executes, but any kind of query, eg `SELECT * FROM version;`, fails.

But, just running the statement that defines the view works fine, eg:
```
select
`base`.`version`.`VersionMetadata` AS `VersionMetadata`,
`base`.`version`.`VersionDatabase` AS `VersionDatabase`,
`base`.`version`.`Date` AS `Date`
from `base`.`version`
```

Yes, it's just supposed to show the table from the other database.

Also, trying to create a view that depends on another view, that references another database fails with the same error.

**Comment by theme1256** [comment]
I'm running 1.30.2, by the way.

**Comment by max-hoffman** [comment]
Thanks for finding this hole in our testing @theme1256 , we will release the fix in dolt 1.30.3 today. Let us know if you have any other migration blockers. We are happy to offer more general help if you are interested in sharing more about your specific use case.
